File: reader/server/core/backend/src/main/java/com/richodemus/reader/backend/model/Item.java
package com.richodemus.reader.backend.model;

import com.richodemus.reader.dto.ItemId;

import java.time.Duration;
import java.time.LocalDate;
import java.util.Objects;

public class Item {
    private final ItemId id;
    private final String title;
    private final String description;
    private final String uploadDate;
    private final String url;
    private final Duration duration;
    private final long views;

    public Item(ItemId id,
                String title,
                String description,
                String uploadDate,
                String url,
                Duration duration,
                long views) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.uploadDate = uploadDate;
        this.url = url;
        this.duration = duration;
        this.views = views;
    }

    private String toDoubleDigitSeconds(long seconds) {
        final String string = String.valueOf(seconds);
        if (string.length() == 1) {
            return "0" + string;
        }
        return string;
    }

    public ItemId getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getUploadDate() {
        return uploadDate;
    }

    public String getUrl() {
        return url;
    }

    public Duration getDuration() {
        return duration;
    }

    public long getViews() {
        return views;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Item item = (Item) o;
        return Objects.equals(id, item.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }

    public boolean isBefore(Item targetItem) {
        return LocalDate.parse(uploadDate).isBefore(LocalDate.parse(targetItem.getUploadDate()));
    }
}

File: reader/server/core/backend/src/main/kotlin/com/richodemus/reader/backend/model/Item.kt
package com.richodemus.reader.backend.model

import com.richodemus.reader.dto.ItemId
import java.time.Duration
import java.time.LocalDate

class Item(val id: ItemId,
           val title: String,
           val description: String,
           val uploadDate: String,
           val url: String,
           val duration: Duration,
           val views: Long) {
    private fun toDoubleDigitSeconds(seconds: Long): String {
        val string = seconds.toString()
        return if (string.length == 1) {
            "0$string"
        } else string
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val item = o as Item
        return id == item.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    fun isBefore(targetItem: Item?): Boolean {
        return LocalDate.parse(uploadDate).isBefore(LocalDate.parse(targetItem!!.uploadDate))
    }
}
